aten/src/ATen/native/Histogram.cpp

```diff
@@ -6,117 +6,181 @@
 #include <ATen/native/Resize.h>
 
 #include <tuple>
+#include <vector>
+#include <c10/util/ArrayRef.h>
 #include <c10/core/ScalarType.h>
 #include <c10/core/DefaultDtype.h>
 
-/* Implements a numpy-like histogram function running on cpu
- * https://numpy.org/doc/stable/reference/generated/numpy.histogram.html
+/* Implements a numpy-like histogramdd function running on cpu
+ * https://numpy.org/doc/stable/reference/generated/numpy.histogramdd.html
  *
- * - torch.histogram(input, bins, range=None, weight=None, density=False)
- *   input     - tensor containing the input values. The histogram is computed over the flattened values.
- *   bins      - int or 1D tensor. If int, defines the number of equal-width bins. If tensor, defines the
- *               sequence of bin edges including the rightmost edge.
- *   range     - (float, float), optional. Defines the range of the bins.
- *   weight    - tensor, optional. If provided, weight should have the same shape as input. Each value
- *               in input contributes its associated weight towards its bin's result (instead of 1).
+ * - torch.histogramdd(input, bins, range=None, weight=None, density=False)
+ *   input     - tensor with dimensions (N, D), interpreted as N coordinates in a D-dimensional space.
+ *               If a tensor with more than 2 dimensions is passed, all but the last dimension will be flattened.
```

Comment:
This is a much better explanation that NumPy's description of the input, which does not even address this case explicitly

Reply:
NumPy's histogramdd works only on `(N, D)` shapes and throws an error for higher-dimensional shapes:
```
Traceback (most recent call last):
  File "/home/saketh/anaconda3/lib/python3.8/site-packages/numpy/lib/histograms.py", line 1015, in histogramdd
    N, D = sample.shape
ValueError: too many values to unpack (expected 2)
```